beater/pub.go

```diff
@@ -88,13 +82,10 @@ func (p *publisher) drain() {
 // an error is returned.
 // Calling send after Stop will return an error.
 func (p *publisher) Send(batch []beat.Event) error {
-	p.m.RLock()
-	defer p.m.RUnlock()
-	if p.stopped {
-		return errChanneClosed
-	}
 
 	select {
+	case <-p.done:
+		return errChanneClosed
```

Comment:
I think this could panic. Try this:
```
package main

import (
    "fmt"
)

func main() {
    c := make(chan int, 10)
    d := make(chan int, 10)
    close(d)
    close(c)

    for i := 0; i < 10; i++ {
        fmt.Println(i)
        select {
        case <- c:
        case d <- 0:
        }
    }
    fmt.Println("OK")
}
```

Reply:
I don't think that is equivalent. Or at least I am not able to make the apm server panic.
Of course it still might be this is wrong even if i can't reproduce the panic.

Reply:
OK. Did you find anything in the code that indicates this is not equivalent? It's important that we understand whether it's not panicking because we were just unlucky or there could be some subtlety that makes our code different from that example.

Reply:
I don't think is a subtlety, more like the complexity of the apm server is not translatable to that example.
But seeing that Silvia's suggestion works and it uses an attribute that is meant for that already, i'd do it that way. It has a lock, but 1 lock more / 1 lock less doesn't matter much at this point.